### PR TITLE
Add blog post on the Kubernetes feature

### PR DESCRIPTION
**Pull request by weg2g**
Pls review my article for publication in Feb on Kubernetes site.

### CODE REVIEW

**Comment by bradtopol** [comment]
@weg2g Thank you for your contribution. Please sign the CLA so we can accept your PR for review.
/assign @kbarnard10

**Comment by weg2g** [comment]
@bradtopol - I have already submitted & signed sign the CLA.  It may take time to show up on your end.

**Review by sftim — CHANGES_REQUESTED** [review]
Hi @weg2g 

Would you be willing to submit this article in [Markdown](https://commonmark.org/) format rather than HTML?

We use Markdown format for articles, and other content, whereever that's possible.

**Comment by weg2g** [comment]
@sftim- I just converted html doc to markdown one. Thx

File: content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,317 @@
+---
+layout: blog
+title: "Working with PersistentVolume and Secrets for Deploying MySQL on Kubernetes"
+date: 2021-01-25
+slug: work-with-persistent-volume-and-secrets-for-deploying-mysql-on-kubernetes
+---
+
```
Comment (sftim) [inline content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md:7]: Recommended: add an author credit.

File: content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+concepts covered in this article will provide you with good foundation
+for applying them to your next Kubernetes full-stack application
+deployment project.  
+
+Kubernetes Persistent Volume
+----------------------------
+
+Let’s start off by explaining what a **Persistent Volume** (**PV**) is
+and what role does it play in the [Kubernetes
+Pods](https://coding-bootcamps.com/blog/kubernetes-evolution-from-virtual-servers-and-kubernetes-architecture.html).
```
Comment (sftim) [inline content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md:24]: For Pod, I recommend linking to https://kubernetes.io/docs/concepts/workloads/pods/

File: content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+resources (CPU and Memory); whereas, PVCs can request specific size and
+access modes (e.g. read-write or read-only). In short, Pods and PV work
+hand-in-hand to use the system resources in a way that is conducive to
+their operation. 
+
+Kubernetes Secrets
+------------------
+
+Put it simply, we use [Kubernetes
+secrets](https://blockchain.dcwebmakers.com/blog/advance-topics-for-deploying-and-managing-kubernetes-containers.html)
```
Comment (sftim) [inline content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md:48]: For Secret, I recommend linking to https://kubernetes.io/docs/concepts/configuration/secret/

File: content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+in the YAML representation:
+
+      apiVersion: v1
+      kind: Secret
+      metadata: 
+      name: mysecret
+      type: Opaque
+      data: 
+      username: YWRtaW4=
+      password: MWYyZDFlMmU2N2Rm
```
Comment (sftim) [inline content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md:66]: If you want, you can have this syntax-highlighted as YAML

File: content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+variables. The environment variables access these credentials from
+Kubernetes secrets.\
+ For the next step, you can start a Minikube cluster to create
+Kubernetes secrets to store database credentials and deploy the MySQL
+instance.
+
+### Summary
+
+In this article, you learn about how to deploy MySQL on
+[Kubernetes](https://myhsts.org/tutorial-review-of-17-essential-topics-for-mastering-kubernetes.php)
```
Comment (sftim) [inline content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md:291]: Omit this hyperlink. The blog already links to the top of the site.

File: content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md [hunk 6]
```diff
@@ -0,0 +1,317 @@
...
+-   [Advance System Admin Guide- 9 Best Practices for Managing
+    Kubernetes](https://blockchain.dcwebmakers.com/blog/advance-topics-for-deploying-and-managing-kubernetes-containers.html)
+-   [System Admin Guide- What is Kubernetes and how it
+    works](https://coding-bootcamps.com/blog/kubernetes-evolution-from-virtual-servers-and-kubernetes-architecture.html)
+-   [Comprehensive Guide for Migration From Monolithic To Microservices
+    Architecture](https://blockchain.dcwebmakers.com/blog/comprehensive-guide-for-migration-from-monolithic-to-microservices-architecture.html)
+-   [Review of Pod-to-Pod Communications in
+    Kubernetes](https://coding-bootcamps.com/blog/pod-to-pod-communications-in-kubernetes.html)
+-   [Build and Deploy Containerized Applications with Golang on
+    Kubernetes](https://coding-bootcamps.com/blog/build-containerized-applications-with-golang-on-kubernetes.html)
```
Comment (sftim) [inline content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md:315]: I would omit these links in favor of internal hyperlinks.

**Review by sftim — COMMENTED** [review]
This is similar to [Run a Single-Instance Stateful Application](https://kubernetes.io/docs/tasks/run-application/run-single-instance-stateful-application/)

I like the more narrative approach here compared to this task page.

@weg2g would you be willing to make a broader change to line up the blog article and the task to use the same manifests, or at least compatible manifests?

Alternatively, we could migrate https://kubernetes.io/docs/tasks/run-application/run-single-instance-stateful-application/ to a new tutorial page that draws on your work for this blog.

File: content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md [hunk 7]
```diff
@@ -0,0 +1,317 @@
...
+
+
+Deploy MySQL on Kubernetes Using PV and Secrets
+-----------------------------------------------
+
+Following is the Kubernetes manifest for MySQL deployment. We have added
+comments alongside each configuration to make sure that its usage is
+clear to you.
+
+`   `
```
Comment (sftim) [inline content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md:77]: Should this be <tt>```yaml</tt>?

File: content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md [hunk 8]
```diff
@@ -0,0 +1,317 @@
...
+### Summary
+
+In this article, you learn about how to deploy MySQL on
+[Kubernetes](https://myhsts.org/tutorial-review-of-17-essential-topics-for-mastering-kubernetes.php)
+and in doing so you learned three important Kubernetes concepts:
+Persistent Volume, Persistent Volume Claim and Secrets. We also reviewed
+a sample configuration environment for setting the right properties for
+PV, PVC and Secrets on the manifest file. You can extend this article by
+using Kubernetes tools like Minikube to make Kubernetes secrets to store
+database credentials and deploy the MySQL instance.
```
Comment (sftim) [inline content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md:297]: > You can extend this article by using Kubernetes tools like Minikube to make Kubernetes secrets to store database credentials and deploy the MySQL instance.

I don't understand this sentence.

**Comment by zacharysarah** [comment]
@weg2g Thanks for opening this PR.

The proposed language contains numerous grammatical errors and requires significant revision before this PR could merge acceptable.

Given the reference links to a third party blockchain provider, this looks like an article designed to improve the linked site’s SEO. This article resembles a vendor pitch, and per the blog content guide, vendor pitches aren’t allowed.

I recommend the blog maintainers decline to publish this article without significant revisions to grammar and community orientation.

/hold

File: content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md (hunk 1 above)
Comment (weg2g) [reply content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md:7]: I agree...Added

File: content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md (hunk 6 above)
Comment (weg2g) [reply content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md]: I removed all of them

File: content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md (hunk 2 above)
Comment (weg2g) [reply content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md]: I agree

File: content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md (hunk 3 above)
Comment (weg2g) [reply content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md]: I did

File: content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md (hunk 4 above)
Comment (weg2g) [reply content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md:66]: No need

File: content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md (hunk 5 above)
Comment (weg2g) [reply content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md]: I removed it

File: content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md (hunk 8 above)
Comment (weg2g) [reply content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md]: Removed

**Comment by weg2g** [comment]
@sftim - I followed your comments. I am going to add few more paragraphs to make it more narrative and polish its language against any grammar issues. I get back to you in next 7 days. 

@zacharysarah - I removed all links in the article.

File: content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md (hunk 4 above)
Comment (tengqm) [reply content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md:66]: It is actually needed... for example, the line 62, 65, 66 need more indentation to make it a valid YAML.

**Review by mrbobbytables — COMMENTED** [review]
Thanks for your idea, but content wise this seems very much like the tasks/tutorials we have already and is not something I'd aim republish on the blog.

A few examples:
- https://kubernetes.io/docs/tasks/configure-pod-container/configure-persistent-volume-storage/
- https://kubernetes.io/docs/tutorials/stateful-application/mysql-wordpress-persistent-volume/

**Comment by sftim** [comment]
Given the firm opinion in https://github.com/kubernetes/website/pull/26241#pullrequestreview-594872194 - which makes sense to me - I think the next step here would be to close this PR @weg2g 

I think if you want to submit this for publication elsewhere, incorporating feedback so far, that wouldn't be a problem.

**Comment by kbarnard10** [comment]
@weg2g Going to close this out per @mrbobbytables and @sftim's comments.

**Pull request closed without merging**